### DISCUSSION TITLE
Portainer Add On

### DISCUSSION THREAD

**Discussion by gfn256** [alexbelgium/hassio-addons#739] [Q&A]
I'm running Home Assistant on my Raspberry Pi 4 8gb as follows: 
Home Assistant 2023.2.5
Supervisor 2023.01.1
Operating System 9.5
Frontend 20230202.0 - latest

My question is as follows.

If I add this Portainer addon, will it render my system Unsupported/Unhealthy?

The reason I want to install Portainer, is to be able to run a Cups Print Server in a container.

**Comment by alexbelgium** [comment]
Hi, it all depends how it is used :slight_smile: running an external container with portainer gives an unsupported state. But not running the addon itself

So you just need to stop all custom containers before updating HA

You could use this addon though : https://github.com/MaxWinterstein/homeassistant-addons the dev is very nice

**Reply by gfn256** [reply]
Thanks for speedy answer!😊
The suggestion of using MaxWinterstein's addon, in fact I have already done that myself, not with his addon, but with one that he has based himself on, but the problem is that it doesn't work 100% right. (Printers are not announced on the network etc..). MaxWinterstein's  has the same problems as you can see from his GitHub site.
For that reason I want to use a real container for CUPS. Do you think this should be possible with your Portainer addon?
I can live with the fact that I turn the CUPS container off before updating HA.
I got the idea from the following link:
https://community.home-assistant.io/t/setting-up-local-usb-printer-with-cups-via-portainer/204112

**Reply by alexbelgium** [reply]
Well it indeed should work! Also perhaps his addon just needs access to host network. Good luck!

**Reply by gfn256** [reply]
Thanks!😊 You're fast!
Maybe you could see if you could get a cups server working as it should? I think with you're skills you should be able to.
If you search the various HA forums  you will see there is enormous demand for a CUPS server within an HAOS installation.
Many thanks again. 👍

**Reply by alexbelgium** [reply]
I'm sorry i don't use it and wouldn't be able to teoubleshoot it adequatly, it would be much easier to ask @MaxWinterstein to adapt his own addon
